includes/class-wc-stripe-autoloader.php

```diff
@@ -0,0 +1,217 @@
+<?php
+
+/**
+ * Class that implements a basic autoloader for the WooCommerce Stripe payment gateway.
+ *
+ * @since 9.5.0
+ */
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+class WC_Stripe_Autoloader {
+
+	/**
+	 * Cached in-memory class map. Will be populated from {@see get_classmap()}.
+	 *
+	 * @var string[]|null
+	 */
+	private static $classmap = null;
+
+	/**
+	 * Cached in-memory class map for admin code. Will be populated from {@see get_admin_classmap()}.
+	 *
+	 * @var string[]|null
+	 */
+	private static $admin_classmap = null;
+
+	/**
+	 * Tries to autoloads a class based on the classmap from {@see get_classmap()}
+	 * and when {@see is_admin()} is true, we access {@see get_admin_classmap()} as well.
+	 *
+	 * @param string $class The class to autoload.
+	 * @return boolean True if the class was autoloaded, false otherwise.
+	 */
+	public static function autoload( $class ) {
+		// We're not using namespaces, so skip if the class name contains a namespace.
+		if ( str_contains( $class, '\\' ) ) {
+			return false;
+		}
+
+		// Note that we lowercase the class name as PHP class names are case-insensitive.
+		// We intentionally avoid prefix matching because that would require multiple prefix checks for every lookup.
+		// Instead we use key lookups against static classmaps with one-time load costs.
+		$class_lower = strtolower( $class );
+
+		if ( null === self::$classmap ) {
+			self::$classmap = self::get_classmap();
+		}
+
+		if ( isset( self::$classmap[ $class_lower ] ) ) {
+			require self::$classmap[ $class_lower ];
+
+			return true;
```

Comment:
I was about to suggest adding a check to see if the file exists, to avoid fatal errors when there is a mapping to a non-existent file. However, I suppose throwing a fatal error is clearer than logging an error, which could be overlooked.